core/modules/global/stringUtil.h

```diff
@@ -65,5 +65,19 @@ sanitizeName(std::string const& name) {
     return out;
 }
 
+/**
+ * Convert a string to lowercase
+ *
+ * @param name input string to convert to lowercase
+ * @return the converted string
+ */
+inline std::string toLower(std::string const& name) {
```

Comment:
I second the comment that we should not write stuff like this ourself and use boost::algorithm::to_lower(_copy).